File: solutions/347-top-k-frequent-elements/top-k-frequent-elements.ts
function topKFrequent(nums: number[], k: number): number[] {

    const freqMap = new Map<number, number>();
    for(let i = 0; i < nums.length; i += 1) {
        freqMap.set(nums[i], (freqMap.get(nums[i]) || 0) + 1);
    }

    const pq: number[][] = [];
    for(const [num, freq] of freqMap.entries()) {
        pq.push([num, freq]);
    }

    pq.sort((a, b) => b[1] - a[1]);

    while(pq.length > k) {
        pq.pop();
    }

    const res: number[] = [];
    for(let i = 0; i < pq.length; i += 1) {
        res.push(pq[i][0]);
    }
    return res; 
};

File: solutions/347-top-k-frequent-elements/top-k-frequent-elements.py
import heapq
from typing import List

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        freqMap = {}
        for i in range(len(nums)):
            curr_num = nums[i]
            if curr_num not in freqMap:
                freqMap[curr_num] = 1
            else:
                freqMap[curr_num] += 1
        
        max_pq = []
        for num, count in freqMap.items():
            heapq.heappush_max(max_pq, (count, num))
        
        res = []
        for _ in range(k):
            (count, num) = heapq.heappop_max(max_pq)
            res.append(num)

        return res